pkg/gardenlet/operation/botanist/monitoring.go

```diff
@@ -376,3 +457,67 @@ func (b *Botanist) getMonitoringComponents() []component.MonitoringComponent {
 
 	return monitoringComponents
 }
+
+// TODO(rfranzke): Remove this function after v1.100 has been released.
+func normalizeAlertingRules(input string) []string {
+	var cleanedLines []string
+	for _, line := range strings.Split(input, "\n") {
+		if strings.Contains(line, ": |") {
+			continue
+		}
+		if len(line) >= 2 && line[:2] == "  " {
+			cleanedLines = append(cleanedLines, line[2:])
+		} else {
+			cleanedLines = append(cleanedLines, line)
+		}
+	}
+
+	var blocks []string
+	for _, group := range strings.Split(strings.Join(cleanedLines, "\n"), "groups:") {
+		group = strings.TrimSpace(group)
+		if group != "" {
+			blocks = append(blocks, "groups:\n"+group)
+		}
+	}
+
+	return blocks
+}
+
+// TODO(rfranzke): Remove this function after v1.100 has been released.
+func normalizeScrapeConfigs(input string) []string {
+	cleanLines := func(lines []string) []string {
+		var cleanedLines []string
+		for _, line := range lines {
+			line = strings.TrimPrefix(line, "- ")
+			if len(line) >= 2 && line[:2] == "  " {
+				cleanedLines = append(cleanedLines, line[2:])
+			} else {
+				cleanedLines = append(cleanedLines, line)
+			}
+		}
+		return cleanedLines
+	}
+
+	var (
+		cleanedBlocks   []string
+		blockStartIndex = 0
+		lines           = strings.Split(input, "\n")
+	)
+
+	for i := 0; i < len(lines); i++ {
+		if strings.HasPrefix(lines[i], "- ") {
+			if i > blockStartIndex {
+				block := strings.Join(cleanLines(lines[blockStartIndex:i]), "\n")
+				cleanedBlocks = append(cleanedBlocks, block)
+			}
+			blockStartIndex = i
+		}
+	}
+
+	if blockStartIndex < len(lines) {
+		block := strings.Join(cleanLines(lines[blockStartIndex:]), "\n")
+		cleanedBlocks = append(cleanedBlocks, block)
+	}
+
+	return cleanedBlocks
+}
```

Comment:
Similar to the function above, it does some magic string manipulation without any mention why this is required. As above I can accept it as long as this function gets removed soon as stated above.